src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -680,6 +684,73 @@ impl CatalogState {
         Ok(BuiltinTableUpdate { id, row, diff })
     }
 
+    pub fn pack_aws_connection_update(
+        &self,
+        connection_id: GlobalId,
+        aws_config: &AwsConfig,
+        diff: Diff,
+    ) -> Result<BuiltinTableUpdate, Error> {
+        let id = self.resolve_builtin_table(&MZ_AWS_CONNECTIONS);
+
+        let mut access_key_id: Option<&str> = None;
+        let mut access_key_id_secret_id = None;
+        let mut assume_role_arn: Option<&str> = None;
+        let mut assume_role_session_name: Option<&str> = None;
+        let mut principal: Option<&str> = None;
+        let mut external_id = None;
+        let mut example_trust_policy = None;
+
+        match &aws_config.auth {
+            AwsAuth::Credentials(credentials) => match &credentials.access_key_id {
+                StringOrSecret::String(access_key) => access_key_id = Some(access_key),
+                StringOrSecret::Secret(secret_id) => {
+                    access_key_id_secret_id = Some(secret_id.to_string())
+                }
+            },
+            AwsAuth::AssumeRole(assume_role) => {
+                assume_role_arn = Some(&assume_role.arn);
+                assume_role_session_name = assume_role.session_name.as_deref();
+                principal = Some(&assume_role.mz_principal.arn);
+                external_id = Some(
+                    assume_role
+                        .mz_principal
+                        .get_external_id(connection_id.to_string()),
+                );
+                example_trust_policy = Some(
+                    Jsonb::from_serde_json(
+                        assume_role
+                            .mz_principal
+                            .get_aws_example_trust_policy(connection_id.to_string()),
+                    )
+                    .expect("valid json expected"),
+                );
+            }
+        }
+        let mut row = Row::default();
+        let mut row_packer = row.packer();
+        row_packer.push(Datum::String(&connection_id.to_string()));
+        row_packer.push(Datum::from(
+            aws_config.endpoint.as_ref().map(String::as_str),
+        ));
+        row_packer.push(Datum::from(aws_config.region.as_ref().map(String::as_str)));
+        row_packer.push(Datum::from(access_key_id));
+        row_packer.push(Datum::from(
+            access_key_id_secret_id.as_ref().map(String::as_str),
+        ));
+        row_packer.push(Datum::from(assume_role_arn));
+        row_packer.push(Datum::from(assume_role_session_name));
+        row_packer.push(Datum::from(principal));
+        row_packer.push(Datum::from(external_id.as_ref().map(String::as_str)));
+
+        if let Some(policy) = example_trust_policy {
+            row_packer.push(policy.into_row().into_element())
+        } else {
+            row_packer.push(Datum::Null)
+        }
```

Comment:
I had tried this and other variations, but I was getting borrow checker errors and the policy field was the only reason that I couldn't use `pack_slice`. Let me give it a try again, will paste the error otherwise.

Reply:
This is the error
```
 --> src/adapter/src/catalog/builtin_table_updates.rs:785:42
    |
785 |             example_trust_policy.map(|p| p.into_row().into_element()),
    |                                          ------------^^^^^^^^^^^^^^^
    |                                          |
    |                                          returns a value referencing data owned by the current function
    |                                          temporary value created here

```

Reply:
resolved with the latest commit